### DISCUSSION TITLE
Upgrading to v.2.0: Database config/setup using existing .env info = "access violation:1071:

### DISCUSSION THREAD

**Discussion by NeadReport** [Kovah/LinkAce#833] [Support]
_Database could not be configured. Please check your connection details. Details: SQLSTATE[42000]: Syntax error or access violation: 1071 Specified key was too long; max key length is 1000 bytes (Connection: mysql, SQL: alter table `permissions` add unique `permissions_name_guard_name_unique`(`name`, `guard_name`))_

**Comment by Kovah** [comment]
Hi, thanks for testing LinkAce 2!
Could you please share more details about the database you are running? Type and version would be needed to test this.

**Reply by NeadReport** [reply]
Yes, sorry for the delay. I actually worked on this with the support group at my domain provider. 
Here were their suggestions which I followed and they looked at and confirmed.

_"This error is usually caused when importing data from a MySQL 8 database to a MySQL 5 database as the varchar limit is **191** characters for utf8mb4 encoding in MySQL 5 and **255** in MySQL 8. The easiest way to resolve this problem would be to migrate your MySQL 5 database to MySQL 8."_

I migrated my database from MySQL 5 to MySQL 8 and then we found out the port also changes from 3306 to 3308. I did this and it updated on the install wizard DB form.

The error continues however and the support team thinks it is a bug with the installer.

(this is Sam, btw from the Cloudhiker Discord server :)

**Reply by Kovah** [reply]
Oh hey!
Thanks for the details. Will have a look later. I guess this can be solved in the database updates running during the upgrade.

**Reply by NeadReport** [reply]
Cool! Thanks. I'll let you know after you finish.

**Reply by Kovah** [reply]
Could you check which storage engine you are using in MySQL for the LinkAce database?

**Comment by NeadReport** [comment]
Hey Kevin,

Any chance you can look into this DB issue? I can play around with it this weekend. Right now I'm kinda dead in the water.

Thanks! Have great weekend.

Sam
E-mail is an impersonal medium. This message is intended to be brief, not unfriendly.

Sent with [Proton Mail](https://pr.tn/ref/X0672A01A08G) secure email.

**Reply by Kovah** [reply]
Not yet, sorry. Too much to do atm.

**Comment by NeadReport** [comment]
How is it you release a beta update, want testing feedback, but are too busy to look into these issues?

That should have been a disclaimer when you released the beta, right?

Sam
E-mail is an impersonal medium. This message is intended to be brief, not unfriendly.

Sent with [Proton Mail](https://pr.tn/ref/X0672A01A08G) secure email.

**Reply by Kovah** [reply]
I don't think that there's any obligation for me to work on LinkAce every day or as soon as some issue pops up. I do everything on LinkAce entirely in my free time, and I would like you to respect that I take time when I can and want.

**Comment by NeadReport** [comment]
I support you via CloudHiker.

Sam
E-mail is an impersonal medium. This message is intended to be brief, not unfriendly.

Sent with [Proton Mail](https://pr.tn/ref/X0672A01A08G) secure email.

**Comment by NeadReport** [comment]
I rolled this back to the last stable version I had before the beta update. So that would be MySQL 5 on port 3306
⟶ utf8mb4_unicode_ci ⟵ is that what you need?

Sam
E-mail is an impersonal medium. This message is intended to be brief, not unfriendly.

Sent with [Proton Mail](https://pr.tn/ref/X0672A01A08G) secure email.

**Comment by NeadReport** [comment]
The MySQL database uses the MyISAM storage engine. The database collation is set to utf8mb4_bin, and the table collation is set to utf8mb4_unicode_ci.

Sam
E-mail is an impersonal medium. This message is intended to be brief, not unfriendly.

Sent with [Proton Mail](https://pr.tn/ref/X0672A01A08G) secure email.

**Reply by Kovah** [reply]
Please try changing the storage engine from MyISAM to InnoDB. This was suggested as a fix in many threads for this particular error.

**Comment by NeadReport** [comment]
Well, that took some doing, but it appears I have updated to the 2.0 beta version.
[image.png]
Thanks for the DB info, seems that was the fix.
I'll keep you posted via Github on any bugs.

Sam
E-mail is an impersonal medium. This message is intended to be brief, not unfriendly.

Sent with [Proton Mail](https://pr.tn/ref/X0672A01A08G) secure email.
